### ISSUE TITLE
OS X support status?

### ISSUE THREAD

**Issue by UniqMartin** [Openwsman/openwsman#89]
`openwsman` is packaged by the Homebrew package manager for OS X, but that package hasn't been maintained for a while. One issue is that this package is still using the deprecated autotools build system, resulting in a broken/incomplete build and impacting the build of the `wsmancli` package. (See Homebrew/legacy-homebrew#44047 for details.)

Not being a `openwsman` or `wsmancli` user, I attempted to switch the build to the suggested CMake build system. However, I quickly noticed that this is making some Linux-specific assumptions, thus I didn't get very far. What is the status of OS X support? If it's basically unsupported, the mentioned packages will need to be removed from Homebrew.

Issues I noticed while trying to make the build work (non-comprehensive list):

- Non-portable [attempt to infer things directly from `libssl.so`](https://github.com/Openwsman/openwsman/blob/4e54d3ee9d5830154a94e6cdc883c43bcc860e73/src/server/CMakeLists.txt#L16). Both the assumed directory of that library and its file extension happen to be different on OS X.
- `wsman_curl_client_transport` uses OpenSSL includes (and functions), [but doesn't specify it in its dependencies](https://github.com/Openwsman/openwsman/blob/4e54d3ee9d5830154a94e6cdc883c43bcc860e73/src/lib/CMakeLists.txt#L53-L57), causing the link to fail with unresolved symbols.

Thanks!

**Comment by kkaempf** [comment]
Right, the `libssl.so` hack is Linux specific. Whats needed in OS X here ?

The OpenSSL dependency for `wsman_curl_client_transport` has been added. Thanks for pointing this out.

**Comment by UniqMartin** [comment]
> Right, the `libssl.so` hack is Linux specific. Whats needed in OS X here ?

Not being exactly familiar with the project, I read from the source that this is needed for dynamically loading the SSL library at run time. Any reason why it cannot be linked during build time, thus taking advantage of the linker, linker paths an the platform-independent resolution of a `-lssl`?

If you really need or want to stick with run-time linking, on OS X this needs to be a full path (instead of a relative path that works on Linux). In the case of Homebrew, OpenSSL can reside in various locations and the only way to pick up the right one is to make the path user-configurable (so that it can be supplied by Homebrew; instead of `${LIB_INSTALL_DIR}/libssl.so` where the assumption seems to be that OpenSSL and `openwsman` are going to live in the same tree).

Even for Linux, you most likely don't want to blindly follow the symlink, but instead extract the `SONAME` from that library like the (build time) linker would do. (It can be read with `objdump -p`, but maybe there's also a more elegant way for that, maybe even one already provided by CMake.) The equivalent of the `SONAME` on OS X is the so-called “install name” (or “dylib ID”). It can be determined with `otool -D` like so:

```
$ otool -D /opt/homebrew/opt/openssl/lib/libssl.dylib
/opt/homebrew/opt/openssl/lib/libssl.dylib:
/opt/homebrew/opt/openssl/lib/libssl.1.0.0.dylib
```

(That's on my system. Most users will have `/usr/local` in place of `/opt/homebrew`.) But again, that would be much simpler if you could just rely on the linker.

> The OpenSSL dependency for `wsman_curl_client_transport` has been added. Thanks for pointing this out.

Thanks!

**Comment by kkaempf** [comment]
Loading of `libssl` is done in [shttpd](https://github.com/Openwsman/openwsman/blob/master/src/server/shttpd/config.c#L95), a third-party component copied 1:1 to openwsman. I'm not going to touch this :wink: 

So we can't rely on the linker here but need the full path to libssl.

**Comment by UniqMartin** [comment]
> Loading of `libssl` is done in shttpd, a third-party component copied 1:1 to openwsman. I'm not going to touch this :wink:

Alright, it's not what I see as the best possible solution, but there are alternatives, so that's not a road block. As I outlined in my comment above, its perfectly possible to make that work with OS X, too. We just have to figure out the right `SSL_LIB` in CMake and then pass than on when building that file.

So how do we move on from here? I think getting feedback from you on #92 opened by a Homebrew co-maintainer of mine would be great, as it takes some first steps making the CMake build system more friendly towards OS X. Would be also great to get #90 merged. Thanks!

**Comment by kkaempf** [comment]
#90 is merged, #92 needs to be splitted into separate issues (resp. PRs). I'm also open to replace `shttpd` with a newer version (if exists). See https://github.com/Openwsman/openwsman/pull/86#issuecomment-218386050 :-/

**Comment by kkaempf** [comment]
`shttpd` is at http://docs.huihoo.com/shttpd/. Latest version is 1.42 (from 2008).
